app/actions/remote/notifications.ts

```diff
@@ -180,6 +180,13 @@ export const backgroundNotification = async (serverUrl: string, notification: No
 };
 
 export const openNotification = async (serverUrl: string, notification: NotificationWithData) => {
+    // Wait for initial launch to kick in if needed
+    await new Promise((r) => setTimeout(r, 500));
+
+    if (EphemeralStore.getProcessingNotification() === notification.identifier) {
+        return {};
+    }
```

Comment:
Definitely not a fan, why would we need to wait for half a second? What is causing the race condition here? This looks like a random hack 🤣

Reply:
If I remember correctly somewhere around this flow we wait for the home screen to be in the stack, perhaps we should move that here instead?

Reply:
The problem is that that is on the NavigationStore... which is not purged when we hit back, and it is not purged until initialize is executed, that I think happen after the event is handled (or at least may happen). So we would still be open to some form of race condition.

May be enough, since we have several awaits in `openNotification` that we do not have on `initialize`, but 0/5.